### PR TITLE
ADDON-56381: Added support for YAML file

### PR DESCRIPTION
**Pull request by tbalar-splunk**
- Added support to use yaml format. So, **we can use globalConfig.json or globalConfig.yaml file**

### CODE REVIEW

File: src/main/webapp/util/script.js [hunk 1]
```diff
@@ -14,9 +14,20 @@ export function getBuildDirPath() {
...
+
+function loadYAMLFile() {
+    const yaml = require('js-yaml');
+    return axios.get(`${getBuildDirPath()}/globalConfig.yaml`).then((res) => {
+        return typeof res.data === 'object' ? res.data : yaml.load(res.data);
+    });
+}
+
+export function loadGlobalConfig() {
+    // Get the configuraiton json file in sync mode
```
Comment (artemrys) [inline src/main/webapp/util/script.js:31]: This is out of date after the change.

File: src/main/webapp/util/script.js [hunk 2]
```diff
@@ -14,9 +14,20 @@ export function getBuildDirPath() {
...
     });
 }
+
+function loadYAMLFile() {
+    const yaml = require('js-yaml');
+    return axios.get(`${getBuildDirPath()}/globalConfig.yaml`).then((res) => {
+        return typeof res.data === 'object' ? res.data : yaml.load(res.data);
+    });
+}
+
```
Comment (artemrys) [inline src/main/webapp/util/script.js:29]: Is it possible to add a unit test for this behaviour? It seems to be quite isolated from the source code, so should not very complicated to do.

**Review by artemrys — CHANGES_REQUESTED** [review]

**Comment by artemrys** [comment]
Also, please use conventional commit https://www.conventionalcommits.org/en/v1.0.0/ and sign CLA and Code of Conduct.

File: src/main/webapp/util/script.js (hunk 2 above)
Comment (tbalar-splunk) [reply src/main/webapp/util/script.js:29]: The front-end code does not have a unit test. I intend to include some test cases in the backend code.

**Comment by tbalar-splunk** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/main/webapp/util/script.js (hunk 1 above)
Comment (tbalar-splunk) [reply src/main/webapp/util/script.js]: Done

**Comment by tbalar-splunk** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Review by artemrys — APPROVED** [review]

**Pull request merged**